aten/src/ATen/native/RNN.cpp

```diff
@@ -1445,55 +1446,58 @@ std::tuple<Tensor, Tensor, Tensor> lstm(
   TORCH_CHECK(hx.size() == 2, "lstm expects two hidden states");
-  if (at::cudnn_is_acceptable(_input)) {
-    Tensor output, hy, cy;
-    lstm_cudnn_stub(_input.device().type(), output, hy, cy, _input, hx, _params, has_biases,
-            num_layers, dropout_p, train, bidirectional, batch_first);
-    return std::make_tuple(std::move(output), std::move(hy), std::move(cy));
-  }
-#ifdef USE_MPS
-  if (_input.is_mps() && (mps::is_macos_13_or_newer() || num_layers == 1)) {
-    std::tuple<Tensor, Tensor, Tensor, Tensor, Tensor, Tensor> output = at::_lstm_mps(_input, hx, _params, has_biases,
-            num_layers, dropout_p, train, bidirectional, batch_first);
-    std::tuple<Tensor, Tensor, Tensor> return_values = std::make_tuple(std::get<0>(output), std::get<1>(output), std::get<2>(output));
-    return return_values;
-  } else if (_input.is_mps()) {
-    TORCH_WARN_ONCE("Native multi-layer LSTM support in MPS available only on MacOS 13 onwards.",
-                    " Falling back to LSTMCell iteration.",
-                    " This may have performance implications.");
-  }
-#endif
   // if cells are of different size, that means projections are used
   bool has_projections = (hx[0].sym_size(2) != hx[1].sym_size(2));
-  if (use_miopen(_input, dropout_p)) {
-    if (!has_projections) {
+  // Opt for slow path if we are dealing with subclassed tensor.
+  bool any_subclass_like = at::isTensorSubclassLike(_input) || at::areAnyTensorSubclassLike(hx) || at::areAnyTensorSubclassLike(_params);
+  if (!any_subclass_like) {
+    if (at::cudnn_is_acceptable(_input)) {
       Tensor output, hy, cy;
-      lstm_miopen_stub(_input.device().type(), output, hy, cy, _input, hx, _params, has_biases,
-                num_layers, dropout_p, train, bidirectional, batch_first);
+      lstm_cudnn_stub(_input.device().type(), output, hy, cy, _input, hx, _params, has_biases,
+              num_layers, dropout_p, train, bidirectional, batch_first);
       return std::make_tuple(std::move(output), std::move(hy), std::move(cy));
-    } else {
-      TORCH_WARN_ONCE(
-          "LSTM with projections is not supported with MIOpen. Using default implementation.");
     }
-  }
-
-  if (use_mkldnn(_input, _params, hx)) {
-    if (!has_projections) {
-      if (hx[0].unsafeGetTensorImpl()->has_symbolic_sizes_strides()) {
-        TORCH_WARN_ONCE(
-          "LSTM with symbolic sizes and strides is not supported with oneDNN. Using default implementation.");
-      } else {
+#ifdef USE_MPS
+    if (_input.is_mps() && (mps::is_macos_13_or_newer() || num_layers == 1)) {
+      std::tuple<Tensor, Tensor, Tensor, Tensor, Tensor, Tensor> output = at::_lstm_mps(_input, hx, _params, has_biases,
+              num_layers, dropout_p, train, bidirectional, batch_first);
+      std::tuple<Tensor, Tensor, Tensor> return_values = std::make_tuple(std::get<0>(output), std::get<1>(output), std::get<2>(output));
+      return return_values;
```

Comment:
```suggestion
      return std::move(return_values);
```
But why even assign it here and not just return the r value from std::make_tuple?

Reply:
I have just wrapped the original code here. I think the above change should work, will make it in the next patch. Thanks!